app/services/aws/credentials_service.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Aws
+  class CredentialsService
+    class << self
+      def call(params = {})
+        new(params).call
+      end
+    end
+
+    attr_reader :params
+
+    ROLE_SESSION_NAME = '3scale-porta'
+
+    def initialize(params)
+      @params = params
+    end
+
+    # @return [Hash] Depending on the available connections, this method may return:
+    # { credentials: Aws::STS::Types::AssumeRoleWithWebIdentityResponse } if STS credentials are available;
+    # { web_identity_token_file: 'foo', role_name: 'bar' } if the values are present;
+    # An empy Hash if no configurations are available;
+    def call
+      if sts_params.any? && sts_params.all? { |_key, value| value.present? } && web_identity_token_file_exists?
+        { credentials: sts_credentials }
+      else
+        aws_credentials
```

Comment:
small naming suggestion
```suggestion
        iam_credentials
```

Reply:
Much better name. Changed in bce3c14e5. Thank you!